### ISSUE TITLE
time: unbalanced quotation in Parse error message

### ISSUE THREAD

**Issue by adonovan** [golang/go#45391]
See https://play.golang.org/p/z4GVpb8RQCY:
```
	// time (underlying problem)
	_, err := time.Parse(`"`+time.RFC3339+`"`, "0")
	fmt.Println(err) // parsing time "0" as ""2006-01-02T15:04:05Z07:00"": cannot parse "0" as """

	// encoding/json (how I encountered the error)
	err = json.Unmarshal([]byte(`0`), new(time.Time))
	fmt.Println(err) // parsing time "0" as ""2006-01-02T15:04:05Z07:00"": cannot parse "0" as """
```

The error message emitted by time.Parse contains unbalanced quotation marks, presumably because it uses its own trivial quote function to avoid a cycle-forming dependency on strconv. Ideally its quote function would be a little smarter. (Perhaps time.Parse could also avoid redundantly mentioning "parse" and "0" twice in the error message.)

**Comment by seankhliao** [comment]
The error message looks like it makes sense, it's trying to parse the input as `"2006-01-02T15:04:05Z07:00"` (with the double quotes) and it chokes on missing the leading `"`.

I guess it could us fmt.Sprintf("%q", s) instead to get `"\""`?

**Comment by adonovan** [comment]
The time package cannot use %q (strconv.Quote), but a simple improvement would be to use backticks to quote strings that contain double-quotes.

**Comment by SimonAlling** [comment]
For the record, the fix in 1271e9a9ccfdb0906ecf69d2047ad3b470eeca02 caused some of our tests that passed with Go 1.16 to fail with Go 1.17 because they were written like this:

```go
{
	description: "message with invalid timestamp",
	message:     plainMessage(`{"Value":42,"Timestamp":"HALLOJ"}`),
	expectationOnError: errtest.And(
		errtest.HasMessageContaining(`Unknown message format: parsing time ""HALLOJ""`),
		errtest.HasMessageContaining(`cannot parse "HALLOJ""`),
	),
},
```

Test output:

```
Error message should contain

  Unknown message format: parsing time ""HALLOJ""

Actual error:

  Unknown message format: parsing time "\"HALLOJ\"" as "\"2006-01-02T15:04:05Z07:00\"": cannot parse "HALLOJ\"" as "2006"
```

Not meant as a criticism, just a side note that perhaps someone may find interesting. :slightly_smiling_face:
